From b43d76f87394ea8fbb8d2fc48b94572dc34e9931 Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca.weiss@fairphone.com>
Date: Tue, 14 Nov 2023 22:16:39 +0100
Subject: [PATCH 05/11] Compile with C++17

Some headers in mir include C++17 types such as std::optional so we need
to compile against C++17 also.

In file included from /usr/include/miral/miral/minimal_window_manager.h:21,
                 from /usr/include/qtmir/qtmir/windowmanagementpolicy.h:21,
                 from /home/pmos/build/src/lomiri-0.1.4/src/WindowManagementPolicy.h:20,
                 from /home/pmos/build/src/lomiri-0.1.4/src/LomiriApplication.cpp:38:
/usr/include/miral/miral/window_manager_tools.h:162:72: error: 'optional' in namespace 'std' does not name a template type
  162 |     auto window_to_select_application(const Application) const -> std::optional<Window>;
      |                                                                        ^~~~~~~~
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6a3b43d4d..ea32f07e6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -141,7 +141,7 @@ endif()
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-permissive -pedantic -Wall -Wextra")
 
-set (CMAKE_CXX_STANDARD 14)
+set (CMAKE_CXX_STANDARD 17)
 
 if ("${CMAKE_BUILD_TYPE}" STREQUAL "release" OR "${CMAKE_BUILD_TYPE}" STREQUAL "relwithdebinfo")
     option(Werror "Treat warnings as errors" ON)
-- 
2.43.0

